packages/core/src/middleware/koa-spa-proxy.ts

```diff
@@ -0,0 +1,59 @@
+import fs from 'fs/promises';
+import path from 'path';
+
+import { MiddlewareType } from 'koa';
+import proxy from 'koa-proxies';
+import { IRouterParamContext } from 'koa-router';
+import serveStatic from 'koa-static';
+
+import { isProduction, MountedApps } from '@/env/consts';
+
+export default function koaSpaProxy<StateT, ContextT extends IRouterParamContext, ResponseBodyT>(
+  packagePath = 'ui',
+  port = 5001,
+  prefix = ''
+): MiddlewareType<StateT, ContextT, ResponseBodyT> {
+  type Middleware = MiddlewareType<StateT, ContextT, ResponseBodyT>;
+
+  const distPath = path.join('..', packagePath, 'dist');
+
+  const spaProxy: Middleware = isProduction
+    ? serveStatic(distPath)
+    : proxy('*', {
+        target: `http://localhost:${port}`,
+        changeOrigin: true,
+        logs: true,
+        rewrite: (requestPath) => {
+          // Static files
+          if (requestPath.includes('.')) {
+            return '/' + path.join(prefix, requestPath);
+          }
+
+          // In-app routes
+          return requestPath;
+        },
```

Comment:
Should we add more tests to cover these lines?

Reply:
yes

Reply:
after some investigation the ROI is low here, since we need to mock a request, and the proxy is for dev only. add LOG-1748